AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSAnalyticsTransmissionTarget.m

```diff
@@ -37,10 +56,53 @@ - (MSAnalyticsTransmissionTarget *)transmissionTargetForToken:(NSString *)token
   // Look up for the token in the dictionary, create a new transmission target if doesn't exist.
   MSAnalyticsTransmissionTarget *target = self.childTransmissionTargets[token];
   if (!target) {
-    target = [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:token];
+    target = [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:token parentTarget:self];
     self.childTransmissionTargets[token] = target;
   }
   return target;
 }
 
+- (BOOL)isEnabled {
+  @synchronized([MSAnalytics sharedInstance]) {
+
+    // Get isEnabled value from persistence.
+    // No need to cache the value in a property, user settings already have their cache mechanism.
+    NSNumber *isEnabledNumber = [self.storage objectForKey:self.isEnabledKey];
+
+    // Return the persisted value otherwise it's enabled by default.
+    return (isEnabledNumber) ? [isEnabledNumber boolValue] : YES;
+  }
+}
+
+- (void)setEnabled:(BOOL)isEnabled {
+  @synchronized([MSAnalytics sharedInstance]) {
+    if (self.isEnabled != isEnabled) {
+
+      // Don't enable if the immediate parent is disabled.
+      if (isEnabled && ![self isAncestorEnabled]) {
+        MSLogWarning([MSAnalytics logTag],
+                     @"Can't enable; parent transmission target and/or Analytics service is disabled.");
+        return;
+      }
+
+      // Persist the enabled status.
+      [self.storage setObject:@(isEnabled) forKey:self.isEnabledKey];
+    }
+
+    // Propagate to nested transmission targets. TODO Find a more effective approach.
```

Comment:
You can remove TODO here. As @jwallra mentioned, it would be O(n) for the best case.

Reply:
Done, it was more on the memory usage then the complexity but it should be fine the way it.